Automattic/delphin
# Search input

This pull request adds components for making a custom search input with tokenisation:

<img width="803" alt="screen shot 2016-05-18 at 3 19 46 pm" src="https://cloud.githubusercontent.com/assets/1130674/15376853/8cddde28-1d0c-11e6-95d8-90723e0420c4.png">

It does not match all of the features of the prototype's search, but makes it possible to add/remove keywords, as well as replace them with a synonym.
#### Testing instructions
1. Run `git checkout add/search-input` and start your server
2. Assert that the following works as expected, and that the results update whenever a keyword is changed:
- Open the [`Search` page](http://delphin.localhost:1337/search/?q=hello)
- Check that indeed `hello` is populated as the first token
- Add additional keywords.
- Remove keywords by pressing `backspace` in an empty input box.
- Remove keywords by clicking the chevron by a keyword, then clicking the trash icon.
- Click the chevron by a keyword and click one of the synonyms.
- Add a domain.
- Remove the domain by clicking the `x` next to it.
#### Reviews
